BMM.Core/Implementations/Region/CultureInfoRepository.cs

```diff
@@ -1,53 +1,34 @@
-using System.Collections.Generic;
 using System.Globalization;
-using System.Linq;
+using BMM.Core.Constants;
+using BMM.Core.Implementations.FirebaseRemoteConfig;
 using BMM.Core.Implementations.Region.Interfaces;
-using BMM.Core.Models.Regions;
+using Newtonsoft.Json;
 
 namespace BMM.Core.Implementations.Region
 {
     public class CultureInfoRepository : ICultureInfoRepository
     {
-        private const string UnknownCultureEnglishName = "Unknown language";
-        private IList<CultureInfo> _customCultureInfos;
+        private readonly IFirebaseRemoteConfig _firebaseRemoteConfig;
+        private IList<CultureInfoLanguage> _cultureInfoLanguages;
 
-        public CultureInfoRepository()
+        public CultureInfoRepository(IFirebaseRemoteConfig firebaseRemoteConfig)
         {
-            CreateCustomCultureInfos();
+            _firebaseRemoteConfig = firebaseRemoteConfig;
         }
 
-        public CultureInfo Get(string iso)
-        {
-            try
-            {
-                return new CultureInfo(iso);
-            }
-            catch (CultureNotFoundException)
-            {
-                return GetCustomCultureInfoFor(iso);
-            }
-        }
+        public CultureInfoLanguage GetCultureInfoLanguage(CultureInfo cultureInfo) 
+            => GetCultureInfoLanguage(cultureInfo.Name) ?? CreateDefaultCultureInfoLanguage(cultureInfo.Name);
 
-        private CultureInfo GetCustomCultureInfoFor(string isoCode)
+        public CultureInfoLanguage GetCultureInfoLanguage(string name)
         {
-            var customCulture = _customCultureInfos
-                .FirstOrDefault(c => c.TwoLetterISOLanguageName == isoCode);
-
-            if (customCulture != null)
-                return customCulture;
-
-            return new CustomCultureInfo(isoCode,
-                isoCode,
-                UnknownCultureEnglishName,
-                string.Empty);
+            _cultureInfoLanguages ??= JsonConvert.DeserializeObject<IList<CultureInfoLanguage>>(_firebaseRemoteConfig.CultureInfoLanguages);
+            var cultureInfoLanguage = _cultureInfoLanguages.FirstOrDefault(c => c.Name == name);
+            return cultureInfoLanguage ?? CreateDefaultCultureInfoLanguage(name);
         }
 
-        private void CreateCustomCultureInfos()
+        private CultureInfoLanguage CreateDefaultCultureInfoLanguage(string name)
         {
-            _customCultureInfos = new List<CultureInfo>()
-            {
-                new CustomCultureInfo("yue", "廣東話", "Cantonese", "zh")
-            };
+            return new CultureInfoLanguage(name, name, name);
```

Comment:
can we keep "unknown language" as english name?
It was quite helpful when we added khasi